lib/datadog/statsd/sender.rb

```diff
@@ -96,9 +100,20 @@ def start
 
         # initialize a new message queue for the background thread
         @message_queue = @queue_class.new
-        # start background thread
-        @sender_thread = @thread_class.new(&method(:send_loop))
-        @sender_thread.name = "Statsd Sender" unless Gem::Version.new(RUBY_VERSION) < Gem::Version.new('2.3')
+        begin
+          # start background thread
+          @sender_thread = @thread_class.new do
+            begin
+              send_loop
+            ensure
+              @mx.synchronize { @done = true }
+            end
```

Comment:
If the intention is to handle `ThreadErrors`... I don't think this is doing it. Specifically, if the `ThreadError` is raised by `Thread.new`, then the `ensure` will never run, so `@done` is never going to be set, and every `add` will result in the attempt to recreate the thread.